src/transformStateWithClones.js

```diff
@@ -8,6 +8,37 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const history = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        currentState = {};
+        break;
+      case 'addProperties':
+        currentState = Object.assign({}, currentState, action.extraData);
+        break;
+      case 'removeProperties':
+        currentState = removeProp(currentState, action);
+        break;
+    }
+    history.push(currentState);
```

Comment:
You are pushing the 'currentState' reference into the 'history' array. This will result in all elements of 'history' pointing to the same object. Instead, push a copy of 'currentState' to avoid mutation issues.